docs/spec/interfaces.md

```diff
@@ -0,0 +1,99 @@
+# Interface Contracts
+
+<!-- This image is sourced from https://drive.google.com/open?id=1Mgn0Oz85M9W1Pwfnw3u0OhZmIIxSxcM3lZUEZyPCoB0 -->
+
+![Interface Contracts Overview](images/interface-contracts-overview.svg)
+
+---
+
+## Subscribable
+
+A **Subscribable** resource will emit events that one or more _Subscription_
+can direct to their configured destination.
+
+### Control Plane
+
+A **Subscribable** resource may be referenced in the _from_ field of a
+_Subscription_. The _Subscribable_ resource MUST expose a
+_status.subscribable.channelable_ field (an _ObjectReference_). The resource
+referenced in the _status.subscribable.channelable_ field MUST be a
+_Channelable_ resource; the field MAY refer back to this _Subscribable_ as a
+self referenced resource.
+
+### Data Plane
+
+A **Subscribable** resource produces or forwards events via its
+_status.subscribable.channelable_ resource.
+
+---
+
+## Channelable
+
+A **Channelable** resource contains a list of subscribers and is responsible
+for delivering events to each of them.
```

Comment:
What would be an example here?  Is it a channel?  I am not 100% following this, because below the spec says

> _Sinkable_ is a _Channel_. 

mind shedding some light into this? Thanks